apps/mobile/src/features/Initialization/ui/screens/ChooseBiometricLoginScreen.tsx

```diff
@@ -1,73 +1,79 @@
-import {atoms as a, useTheme} from '@yoroi/theme'
 import {
   parseBoolean,
   useAsyncStorage,
   useMutationWithInvalidations,
 } from '@yoroi/common'
+import {atoms as a, useTheme} from '@yoroi/theme'
 
-import {useFocusEffect} from '@react-navigation/native'
 import {UseSuspenseQueryOptions, useSuspenseQuery} from '@tanstack/react-query'
 import * as React from 'react'
 import {Alert, Text, View} from 'react-native'
-import DeviceInfo from 'react-native-device-info'
+import * as DeviceInfo from 'react-native-device-info'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
-import {useEnableAuthWithOs} from '~/features/Auth/hooks/useEnableAuthWithOS'
+import {useStrings} from '~/kernel/i18n/useStrings'
 import {useMetrics} from '~/kernel/metrics/metricsManager'
 import {Button, ButtonType} from '~/ui/Button/Button'
-import {Space} from '~/ui/Space/Space'
-import {useStrings} from '~/kernel/i18n/useStrings'
 
+import {useAuth} from '~/features/Auth/context/AuthProvider'
+import {usePromise} from '~/hooks/usePromise'
+import {logger} from '~/kernel/logger/logger'
 import {Biometric} from '../illustrations/Biometric'
 
 export const ChooseBiometricLoginScreen = () => {
-  const {atoms: ta, palette: p} = useTheme()
+  const {atoms: ta} = useTheme()
+  const {enableLoginWithHost} = useAuth()
   const strings = useStrings()
   const {track} = useMetrics()
 
-  useFocusEffect(
+  React.useEffect(
     React.useCallback(() => {
       track.onboardingBiometricsPageViewed()
-      // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []),
   )
 
   const {setScreenShown, isPending: isScreenShownLoading} = useSetScreenShown()
-
-  const {enableAuthWithOs, isLoading} = useEnableAuthWithOs({
-    onSuccess: () => setScreenShown(),
+  const {isPending, resolve} = usePromise({
+    promise: enableLoginWithHost,
+    onSuccess: (isOk) => {
```

Comment:
isOk is needed cuz the logic doesn't rely on exceptions, yet the asyncAuthentication with expo can still fail, login.error will send to sentry (if crashes are enabled)